test/test_util.go

```diff
@@ -0,0 +1,174 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// define struct for each object Test Case
+type ObjectTestCase struct {
+	key             string
+	encryption      string
+	expectPassRead  bool
+	expectPassWrite bool
+}
+
+// define struct for each bucket Test Case
+type BucketTestCase struct {
+	testName     string
+	vars         map[string]interface{}
+	objTestCases []ObjectTestCase
+}
+
+// CondCreateObject conditionally creates an S3 Object inside `bucket` with `body` content
+// It is called when we want to test ReadOnly path
+// Creates object with default AWS session that presumably has PutObject permission
+func CondCreateObject(t *testing.T, awsRegion string, bucket string, body string, obj ObjectTestCase) {
```

Comment:
To borrow some terminology from our main Tamr codebase, could call this `MaybeCreateObject`.

Might help with the learning curve for our developers when reading this code.

Reply:
I like it!

Reply:
fixed in https://github.com/hbarros-caylent/terraform-aws-s3/commit/13ddf2df7c2ef25c1b868154342157fb6c9f1d1c